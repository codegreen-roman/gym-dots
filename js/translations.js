const translations = {
    en: {
        greeting: 'Hello World!'
    },
    ee: {
        greetings: 'Tere maailm!'
    },
    ru: {
        greeting: 'Привет мир!'
    },
}

export default translations