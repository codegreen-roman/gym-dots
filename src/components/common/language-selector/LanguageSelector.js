import React from 'react'
import { func, object } from 'prop-types'
import { setLocale } from 'react-redux-i18n'
import { connect } from 'react-redux'
import { map, compose, keys } from 'ramda'
import { withHandlers } from 'recompose'
import { extractTargetValue } from '@utils/helpers'

const option = item => (
    <option key={item} value={item}>{item}</option>
)

const renderOptions = compose(map(option), keys)

export const _LanguageSelector = ({ translations, handleChange }) => (
    <select onChange={handleChange}>
        {renderOptions(translations)}
    </select>
)

_LanguageSelector.propTypes = {
    handleChange: func.isRequired,
    translations: object.isRequired
}

const mapStateToProps = ({ i18n: { translations } }) => ({ translations })

const mapActionsToProps = dispatch => ({
    changeLanguage: compose(dispatch, setLocale)
})

export const LanguageSelector = compose(
    connect(mapStateToProps, mapActionsToProps),
    withHandlers({
        handleChange: ({ changeLanguage }) => compose(changeLanguage, extractTargetValue)
    })
)(_LanguageSelector)
