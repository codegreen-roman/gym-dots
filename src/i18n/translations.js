export const translations = {
    en: {
        greeting: 'Hello World!'
    },
    ee: {
        greeting: 'Tere maailm!'
    },
    ru: {
        greeting: 'Привет мир!'
    },
}
