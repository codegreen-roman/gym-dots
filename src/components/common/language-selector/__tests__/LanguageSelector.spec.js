import React from 'react'
import { shallow } from 'enzyme'
import { _LanguageSelector as LanguageSelector } from '../LanguageSelector'
import { translations } from '../../../../i18n/translations'
import { compose, length, keys, prop } from 'ramda'

const props = {
    translations
}

const setup = props => {
    const actions = {
        handleChange: jest.fn()
    }

    const component = shallow(<LanguageSelector {...props} {...actions} />)

    return {
        component,
        select: component.find('select'),
        actions,
        props
    }
}

describe('LanguageSelector component', () => {
    it('should render', () => {
        const { component } = setup(props)
        expect(component).toMatchSnapshot()
    })

    it('should have options length same as of keys in translations object', () => {
        const { select } = setup(props)
        const app = compose(length, keys, prop('translations'))
        const objLength = app(props)
        expect(select.children().length).toBe(objLength)
    })

    describe('onChange', () => {
        it('should pass a selected value to the onChange handler', () => {
            const value = 'en'
            const { component, actions } = setup(props)
            expect(component).toMatchSnapshot()

            component.find('select').simulate('change', value)

            expect(actions.handleChange).toBeCalledWith(value)
            expect(actions.handleChange).toHaveBeenCalledTimes(1)
            expect(component).toMatchSnapshot()
        })
    })
})
