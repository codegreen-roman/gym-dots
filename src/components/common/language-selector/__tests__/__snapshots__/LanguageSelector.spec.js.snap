// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector component onChange should pass a selected value to the onChange handler 1`] = `
<select
  onChange={[Function]}
>
  <option
    key="en"
    value="en"
  >
    en
  </option>
  <option
    key="ee"
    value="ee"
  >
    ee
  </option>
  <option
    key="ru"
    value="ru"
  >
    ru
  </option>
</select>
`;

exports[`LanguageSelector component onChange should pass a selected value to the onChange handler 2`] = `
<select
  onChange={[Function]}
>
  <option
    key="en"
    value="en"
  >
    en
  </option>
  <option
    key="ee"
    value="ee"
  >
    ee
  </option>
  <option
    key="ru"
    value="ru"
  >
    ru
  </option>
</select>
`;

exports[`LanguageSelector component should render 1`] = `
<select
  onChange={[Function]}
>
  <option
    key="en"
    value="en"
  >
    en
  </option>
  <option
    key="ee"
    value="ee"
  >
    ee
  </option>
  <option
    key="ru"
    value="ru"
  >
    ru
  </option>
</select>
`;
